openeo/rest/connection.py

```diff
@@ -588,6 +617,30 @@ def describe_collection(self, name) -> dict:
         data = self.get('/collections/{}'.format(name)).json()
         return VisualDict("collection", data = data)
 
+    def collection_items(self, name, spatial_extent: Optional[Dict[str, float]] = None, temporal_extent: Optional[List[Union[str, datetime.datetime]]] = None) -> Paginate:
+        """
+        Loads items for a specific image collection.
+        May not be available for all collections.
```

Comment:
It's actually not available for most collections, should we add a clean error in that case?

Reply:
I don't know, I think I followed the route of the rest of the python client where it does not do any "feature-checking" against e.g. the endpoints in "/". But happy to adopt whethever you prefer.

Reply:
Good point ;-)
But you are allowed to do feature checking, or otherwise to catch exceptions. In this case, I guess the server will return a 404 if item metadata is not available, so we could catch that, and translate it into a message that explains to the user that the feature is optional.

Reply:
The request is only done in the (hopefully generic) Paginate iterator so I can't really make that specific unfortunately. There's no request being sent in collection_items. So I could only check against the endpoints in /. But that is something that we could do for all endpoints in general, that doesn't seem too specific for this PR.